relayer/query.go

```diff
@@ -112,6 +112,49 @@ func QueryChannel(ctx context.Context, src *Chain, channelID string) (*chantypes
 		channelID, src.ChainID(), src.ClientID(), src.ConnectionID())
 }
 
+func QueryPortChannel(ctx context.Context, src *Chain, portID string) (*chantypes.IdentifiedChannel, error) {
+	var (
+		srch        int64
+		err         error
+		srcChannels []*chantypes.IdentifiedChannel
+	)
+
+	// Query the latest height
+	if err = retry.Do(func() error {
+		var err error
+		srch, err = src.ChainProvider.QueryLatestHeight(ctx)
+		return err
+	}, retry.Context(ctx), RtyAtt, RtyDel, RtyErr); err != nil {
+		return nil, err
+	}
+
+	// Query all channels for the given connection
+	if err = retry.Do(func() error {
+		srcChannels, err = src.ChainProvider.QueryConnectionChannels(ctx, srch, src.ConnectionID())
+		return err
+	}, retry.Context(ctx), RtyAtt, RtyDel, RtyErr, retry.OnRetry(func(n uint, err error) {
+		src.log.Info(
+			"Failed to query connection channels",
+			zap.String("conn_id", src.ConnectionID()),
+			zap.Uint("attempt", n+1),
+			zap.Uint("max_attempts", RtyAttNum),
+			zap.Error(err),
+		)
+	})); err != nil {
+		return nil, err
+	}
+
+	// Find the specified channel in the slice of all channels
+	for _, channel := range srcChannels {
+		if channel.PortId == portID {
+			return channel, nil
+		}
+	}
+
+	return nil, fmt.Errorf("channel with port{%s} not found for [%s] -> client{%s}@connection{%s}",
```

Comment:
Can we display the list of src channels in this error so that the user can see which ports were available?

Reply:
added